cc/toolchains/llvm/swift_custom_features.bzl

```diff
@@ -88,6 +88,9 @@ swift_no_default_warnings = feature(
                 "-Wno-fortify-source",
                 "-Wno-absolute-value",
                 "-Wno-format",
+                "-Wno-deprecated-declarations",
+                "-Wno-unused-but-set-variable",
+                "-Wno-unused-variable",
```

Comment:
Naïve question -- how are we supposed to know what warning can be removed here? Should we wait for someone from the platform team to ok it?

Reply:
you clarified while I was typing up my review.

Reply:
These are all flags that are enabled by default in clang - so when we're building third party code (hdf5, suiteparse, curl, etc..) - the builds are spammed with a bunch of these warnings from code we don't really control.

The list here is only intended to be silenced in the third party code. In our code we error if these come up (`-Werror`).

@woodfell reviewed the last PR where I introduced this list to start removing noise from the builds. This is just adding on with a few I missed.